src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -4197,4 +4197,56 @@ public static void updatePinsIsRouted(Design design) {
             updatePinsIsRouted(net);
         }
     }
+
+    /**
+     * (Untested) Remove the speicifed pblock from the XDC constraints of the design.
+     * 
+     * This utility will iterate through the lines of the XDC constraints and
+     * remove any lines that contain the specified pblock name and any pblock-related
+     * commands. The commands that are considered pblock-related are:
+     * - add_cells_to_pblock
+     * - create_pblock
+     * - get_pblocks
+     * - resize_pblock
+     * 
+     * This utility is only tested on Vivado-maintained XDC constraints.
+     * 
+     * @param design The design to remove the pblock from.
+     * @param pblockName The name of the pblock to remove.
+     */
+    public static void deletePblock(Design design, String pblockName) {
```

Comment:
I would suggest calling it something more clear:
```suggestion
    public static void removeConstraintsReferencingPBlock(Design design, String pblockName) {
```